admin/server/projects.go

```diff
@@ -280,30 +284,29 @@ func (s *Server) UpdateProject(ctx context.Context, req *adminv1.UpdateProjectRe
 		return nil, status.Error(codes.PermissionDenied, "does not have permission to delete project")
 	}
 
-	// If changing the Github URL, check github app is installed and caller has access on the repo
-	if safeStr(proj.GithubURL) != req.GithubUrl {
-		_, err = s.getAndCheckGithubInstallationID(ctx, req.GithubUrl, claims.OwnerID())
-		if err != nil {
-			return nil, err
-		}
-	}
-
 	githubURL := proj.GithubURL
-	if req.GithubUrl != "" {
-		githubURL = &req.GithubUrl
+	if req.GithubUrl != nil {
+		// If changing the Github URL, check github app is installed and caller has access on the repo
+		if safeStr(proj.GithubURL) != *req.GithubUrl {
+			_, err = s.getAndCheckGithubInstallationID(ctx, *req.GithubUrl, claims.OwnerID())
+			if err != nil {
+				return nil, err
+			}
+			githubURL = req.GithubUrl
+		}
 	}
 
 	opts := &database.UpdateProjectOptions{
 		Name:                 req.Name,
-		Description:          req.Description,
-		Public:               req.Public,
-		ProdBranch:           req.ProdBranch,
+		Description:          valOrDefault(req.Description, proj.Description),
+		Public:               valOrDefault(req.Public, proj.Public),
+		ProdBranch:           valOrDefault(req.ProdBranch, proj.ProdBranch),
 		ProdVariables:        proj.ProdVariables,
 		GithubURL:            githubURL,
 		GithubInstallationID: proj.GithubInstallationID,
 		ProdDeploymentID:     proj.ProdDeploymentID,
-		ProdSlots:            int(req.ProdSlots),
-		Region:               req.Region,
+		ProdSlots:            int(valOrDefault(req.ProdSlots, int64(proj.ProdSlots))),
```

Comment:
Can reduce one level of nesting if putting `int(req.ProdSlots)` instead of `int64(proj.ProdSlots)`

Reply:
I get this error : cannot convert req.ProdSlots (variable of type *int64) to type int.